pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1706,11 +1707,14 @@ static void validateRequest(PinotQuery pinotQuery, int queryResponseLimit) {
         }
         List<Expression> orderByList = pinotQuery.getOrderByList();
         if (orderByList != null) {
+          List<Expression> selectOperands = selectList.get(0).getFunctionCall().getOperands();
```

Comment:
Will it be help if we move the verification for DISTINCT query from here as part of query compilation itself ?

This is too late when the query has been already compiled and converted to PinotQuery.

Take a look at CalciteSqlCompilerTest  - `testSqlDistinctQueryCompilation` It has tests for multiple invalid DISTINCT queries that fail during compilation itself with clear message. Ideally this should be caught there ?

Reply:
@siddharthteotia Pinot supports `ORDER BY` for columns that appear on the selection part. However, the input query is actually the valid SQL query. That's why it seems not to be caught by the Calcite. 

Since this is indeed a valid SQL query while Pinot doesn't support it, it's probably fine to throw the exception during the validation phase for the compiled query.

Reply:
@siddharthteotia NVM. You meant to catch this from `CalciteSqlParser` which is the bridge between Calcite query tree and Pinot's query object. Yes, I agree that we should catch the above here.